src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
@shadowhand 

> are you opposed to having the user open the file with fopen and pass the resource in?

Yes.

The duplication, in my opinion, should be on the inside of probably *few* implementations - if we don't duplicate here, we'll instead be duplicating across *many* implementations of middleware and many other types of client code.

Also, there is no abstraction of `fopen()`, since it's a global function - by avoiding calls to `fopen()` in client code, and instead making calls to an abstract factory interface, we make it much easier to mock such a dependency. You have to consider the fact that the factory does not just provide a dependency (the Stream) but also it *is* a dependency.

I don't know that mocking a resource handle is even possible? So I think we might as well provide the abstraction here - or else, anyone trying to write testable code will likely need to write their own file-handle abstractions.